vehicle/mkv/software/throttle/throttle.c

```diff
@@ -1,738 +1,276 @@
+    gpio_set_mode(DEVIATION_IMPLAUSIBILITY_LED, OUTPUT);
+    gpio_set_mode(OUT_OF_RANGE_IMPLAUSIBILITY_LED, OUTPUT);
+    gpio_set_mode(RJ45_LED1, OUTPUT);
+    gpio_set_mode(RJ45_LED2, OUTPUT);
+
+    gpio_set_mode(SS_ESTOP, INPUT);
+    gpio_set_mode(SS_IS, INPUT);
+    gpio_set_mode(SS_BOTS, INPUT);
+
+    gpio_enable_interrupt(SS_ESTOP);
+    gpio_enable_interrupt(SS_IS);
+    gpio_enable_interrupt(SS_BOTS);
+
+    sei();
+
+    // Initialize motor controller direction
+    m192_command_message.direction_command = MOTOR_ANTICLOCKWISE;
+
+    can_send_m192_command_message();
+
+    while (true) {
+        if (run_1ms) {
+            // External CAN messages
+            if (can_poll_receive_air_control_critical() == 0) {
+                throttle_state.air_state = air_control_critical.state;
+
+                can_receive_air_control_critical();
+
+                if (air_control_critical.state != AIR_STATE_TS_ACTIVE) {
+                    SET_TORQUE_REQUEST(0);
+                    continue;
+                }
+            }
+
+            if (can_poll_receive_brakelight() == 0) {
+                throttle_state.brake_pressed = brakelight.brake_gate;
+                can_receive_brakelight();
+            }
+
+            if (can_poll_receive_dashboard() == 0) {
+                can_receive_dashboard();
+
+                if (!dashboard.ready_to_drive) {
+                    SET_TORQUE_REQUEST(0);
+                    continue;
+                }
+            }
+
+            if (!throttle_state.ready_to_drive) {
+                SET_TORQUE_REQUEST(0);
+                continue;
+            }
+
+            int16_t pos_r = get_throttle_travel(&throttle_r);
+            int16_t pos_l = get_throttle_travel(&throttle_l);
+
+            int16_t pos_min = MIN(pos_r, pos_l);
+            int16_t pos_max = MAX(pos_r, pos_l);
+
+            if (check_brake(pos_min)) {
+                SET_TORQUE_REQUEST(0);
+                continue;
+            }
+
+            bool implausibility = false;
+            implausibility |= check_out_of_range(&pos_r, &pos_l);
+            implausibility |= check_deviation(pos_max, pos_min);
+
+            if (implausibility) {
+                throttle_state.implausibility_fault_counter++;
+
+                if (throttle_state.implausibility_fault_counter
+                    >= IMPLAUSIBILITY_TIMEOUT_MS) {
+                    SET_TORQUE_REQUEST(0);
+                    throttle_state.implausibility_fault_counter = 0;
+                } // else maintain previous throttle request
+
+                continue;
+            } else {
+                // With no implausibility...
+                throttle_state.implausibility_fault_counter = 0;
+
+                // NOTE: If we decide to do a non-linear map, that would go here
+                int16_t torque_request = pos_min * TORQUE_REQUEST_SCALE;
+
+                SET_TORQUE_REQUEST(torque_request);
+            }
+
+            if (throttle_state.send_can) {
```

Comment:
I assume these will send at the freq specified in the yaml (100 Hz), and not at the same speed as the main loop (1000 Hz), right?

Reply:
Yeah this will be set at 100Hz.